mithril-signer/src/runtime/state_machine.rs

```diff
@@ -374,12 +356,43 @@ impl StateMachine {
                 nested_error: Some(e),
             })?;
 
-        Ok(SignerState::Registered { epoch })
+        match self
+            .runner
+            .can_sign_current_epoch()
```

Comment:
I'm wondering if we need to specify the epoch to check here: `can_sign_epoch(epoch)` as it could create some flakiness otherwise?